src/js/containers/keyword/table/ResultsTableContainer.jsx

```diff
@@ -0,0 +1,325 @@
+import { measureTableHeader } from 'helpers/textMeasurement';
+
+import ResultsTableSection from 'components/keyword/table/ResultsTableSection';
+
+const propTypes = {
+    keyword: PropTypes.string,
+    fetchSummary: PropTypes.func
+};
+
+const tableTypes = [
+    {
+        label: 'Contracts',
+        internal: 'contracts'
+    },
+    {
+        label: 'Grants',
+        internal: 'grants'
+    },
+    {
+        label: 'Direct Payments',
+        internal: 'direct_payments'
+    },
+    {
+        label: 'Loans',
+        internal: 'loans'
+    },
+    {
+        label: 'Other',
+        internal: 'other'
+    }
+];
+
+export default class ResultsTableContainer extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            tableType: 'contracts',
+            page: 1,
+            lastPage: true,
+            counts: {},
+            sort: {
+                field: 'Transaction Amount',
+                direction: 'desc'
+            },
+            columns: {},
+            inFlight: false,
+            error: false,
+            results: [],
+            tableInstance: `${uniqueId()}` // this will stay constant during pagination but will change when the keyword or table type changes
+        };
+
+        this.searchRequest = null;
+        this.tabCountRequest = null;
+
+        this.switchTab = this.switchTab.bind(this);
+        this.loadNextPage = this.loadNextPage.bind(this);
+        this.updateSort = this.updateSort.bind(this);
+    }
+
+    componentDidUpdate(prevProps) {
+        if (prevProps.keyword !== this.props.keyword) {
+            // filters changed, update the search object
+            this.loadColumns();
+            this.pickDefaultTab();
+        }
+    }
+
+    pickDefaultTab() {
+        // get the transaction counts for the current filters
+        if (this.tabCountRequest) {
+            this.tabCountRequest.cancel();
+        }
+
+        this.setState({
+            inFlight: true,
+            error: false
+        });
+
+        const filters = {
+            keyword: this.props.keyword
+        };
+
+        this.tabCountRequest = KeywordHelper.performTabCountSearch({
+            filters
+        });
+
+        this.tabCountRequest.promise
+            .then((res) => {
+                this.parseTabCounts(res.data);
+            })
+            .catch((err) => {
+                if (!isCancel(err)) {
+                    this.setState({
+                        inFlight: false,
+                        error: true
+                    });
+
+                    console.log(err);
+                }
+            });
+    }
+
+    parseTabCounts(data) {
+        const transactionCounts = data.results;
+        let firstAvailable = '';
+        let i = 0;
+
+        // Set the first available award type to the first non-zero entry in the
+        while (firstAvailable === '' && i < tableTypes.length) {
```

Comment:
It would be my preference to use a `for` and `break;` instead of a `while` in order to minimize the risk of an infinite loop.